src/Simulation/Simulators/QCTraceSimulator/Circuits/Interface.qs

```diff
@@ -78,6 +78,36 @@ namespace Microsoft.Quantum.Simulation.Simulators.QCTraceSimulators.Implementati
         body intrinsic;
     }
     
-}
+    operation Interface_ApplyIfElse (measurementResult : Result, onResultZeroOp : (Unit => Unit) , onResultOneOp : (Unit => Unit)) : Unit {
+        body intrinsic;
+    }
+
+    operation Interface_ApplyIfElseA (measurementResult : Result, onResultZeroOp : (Unit => Unit is Adj) , onResultOneOp : (Unit => Unit is Adj)) : Unit {
+        body intrinsic;
+    }
+
+    operation Interface_ApplyIfElseC (ctrls : Qubit[], measurementResult : Result, onResultZeroOp : (Unit => Unit is Ctl) , onResultOneOp : (Unit => Unit is Ctl)) : Unit {
+        body intrinsic;
+    }
+
+    operation Interface_ApplyIfElseCA (ctrls : Qubit[], measurementResult : Result, onResultZeroOp : (Unit => Unit is Ctl + Adj) , onResultOneOp : (Unit => Unit is Ctl + Adj)) : Unit {
+        body intrinsic;
+    }
 
+    operation Interface_ApplyConditionally (measurementResults : Result[], resultsValues : Result[], onEqualOp : (Unit => Unit) , onNonEqualOp : (Unit => Unit)) : Unit {
```

Comment:
Could these be `internal`? Not necessary, but might clarify their intended usage.

Reply:
I can make them `internal`, but I am wondering if that makes sense for something in the Interface.qs. I am trying to find out more about this file, if it is supposed to be a reflection of some pre-defined interface for the product, then maybe these should be located elsewhere.

Reply:
The classical control handles are now internal.

Reply:
Oops looks like making these internal invalidates the same test I set out to satisfy. There being internal makes them invisible to the test project. I've made them public again to satisfy the test.

Reply:
Ah, ok. That’s good to know! Sorry for the detour.